build.go

```diff
@@ -6,14 +6,26 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/Masterminds/semver"
 	"github.com/paketo-buildpacks/packit"
 	"github.com/paketo-buildpacks/packit/scribe"
 )
 
-func Build(configParser ConfigParser, logger scribe.Logger) packit.BuildFunc {
+func Build(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser, logger scribe.Logger) packit.BuildFunc {
 	return func(context packit.BuildContext) (packit.BuildResult, error) {
 		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
 
+		projectPath, err := buildpackYMLParser.ParseProjectPath(filepath.Join(context.WorkingDir, "buildpack.yml"))
+		if err != nil {
+			logger.Subprocess("WARNING: error parsing buildpack.yml: %w", err)
```

Comment:
@paketo-buildpacks/dotnet-core-maintainers I don't think it's appropriate to error out here since this parser is only used to display the deprecation warning. On the other hand, the warning may be just as confusing for users. Any preference?

Reply:
I think it's probably fine to error here rather than warn. Hitting this case IRL is quite unlikely. I'd expect this call to fail in detect and/or for another buildpack in the order group.

I'd prefer to error out to avoid adding to this buildpack the eccentricity of warning in a failure case.